enclave/core/sgx/tracee.c

```diff
@@ -123,21 +91,15 @@ oe_result_t _handle_oelog_init(uint64_t arg)
     }
 
     _debug_allowed_enclave = is_enclave_debug_allowed();
-    result = OE_OK;
-done:
-    if (path)
-        oe_free(path);
-
-    return result;
 }
 
-oe_result_t oe_log(log_level_t level, const char* fmt, ...)
+oe_result_t oe_log(oe_log_level_t level, const char* fmt, ...)
 {
     oe_result_t result = OE_FAILURE;
-    oe_log_args_t* args = NULL;
     oe_va_list ap;
     int n = 0;
     int bytes_written = 0;
+    char message[OE_LOG_MESSAGE_LEN_MAX];
```

Comment:
This declares a 2048 byte variable in the stack. Is this safe?

Reply:
I move this to heap memory for now. We really need to come up with a heuristic. Maybe the compiler has a max-stack-size option.

Reply:
Resolved. Thanks for catching this. I fixed the host side only. The same fix is needed for enclaves too.